src/MSBuild/XMake.cs

```diff
@@ -251,6 +251,9 @@ string[] args
             // Initialize OpenTelemetry infrastructure
             OpenTelemetryManager.Instance.Initialize(isStandalone: true);
 
+            // Resets the build completion event, signaling that a new build process is starting.
+            s_buildComplete.Reset();
```

Comment:
Can you actually move this into `Execute()`? For the we're-just-starting-a-Server-request scenario we don't have to do this.

Reply:
Thanks for your suggestions. Updated